### PR TITLE
Protocol contracts page

### PR DESCRIPTION
**Pull request by rupak-nm**
- Added protocols page for `ethereum`, `arbitrum` & `fuji` networks for `contracts`, `cxTokens` & `pods`
- Fixed some `@todo`s in the project

### CODE REVIEW

File: src/components/TableOfContents.scss [hunk 1]
```diff
@@ -0,0 +1,100 @@
+@use "../design-system/utils";
```
Comment (binodnpm) [inline src/components/TableOfContents.scss:1]: I am not convinced if this file is really needed. I believe this whole file does nothing.

File: src/data/tools.ts [hunk 2]
```diff
@@ -1,4 +1,18 @@
-const tools = [
+interface Web3ToolItem {
```
Comment (binodnpm) [inline src/data/tools.ts:1]: Only such interfaces that are Astro props should have inline definition. All other types should go inside `types` folder.

File: src/pages/protocol/arbitrum/contracts.astro [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.ContractArbitrum);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
+  <Seo
```
Comment (binodnpm) [inline src/pages/protocol/arbitrum/contracts.astro:18]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'Neptune Mutual Protocol Contracts'
description: 'View all Neptune Mutual Protocol Contracts'
```

File: src/pages/protocol/arbitrum/pods.astro [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import { Api } from "../../../../types/enum";
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.ContractArbitrum);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
```
Comment (binodnpm) [inline src/pages/protocol/arbitrum/pods.astro:17]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'PODs / Neptune Mutual Protocol Contracts on Arbitrum'
description: 'View all POD contracts of the Neptune Mutual Protocol on Arbitrum'
```

File: src/pages/protocol/ethereum/contracts.astro [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.Contract);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
+  <Seo
```
Comment (binodnpm) [inline src/pages/protocol/ethereum/contracts.astro:18]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'Neptune Mutual Protocol Contracts on Ethereum'
description: 'View all Neptune Mutual Protocol contracts on Ethereum'
```

File: src/pages/protocol/ethereum/cx-tokens.astro [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import { Api } from "../../../../types/enum";
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.Contract);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
```
Comment (binodnpm) [inline src/pages/protocol/ethereum/cx-tokens.astro:17]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'cXTokens / Neptune Mutual Protocol Contracts on Ethereum'
description: 'View all cxToken contracts of the Neptune Mutual Protocol on Ethereum'
```

File: src/pages/protocol/ethereum/pods.astro [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import { Api } from "../../../../types/enum";
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.Contract);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
```
Comment (binodnpm) [inline src/pages/protocol/ethereum/pods.astro:17]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'PODs / Neptune Mutual Protocol Contracts on Ethereum'
description: 'View all POD contracts of the Neptune Mutual Protocol on Ethereum'
```

File: src/pages/protocol/fuji/contracts.astro [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import { Api } from "../../../../types/enum";
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.ContractFuji);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
```
Comment (binodnpm) [inline src/pages/protocol/fuji/contracts.astro:17]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'Neptune Mutual Protocol Contracts on Fuji'
description: 'View all Neptune Mutual Protocol contracts on Fuji'
```

File: src/pages/protocol/fuji/cx-tokens.astro [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import { Api } from "../../../../types/enum";
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.ContractFuji);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
```
Comment (binodnpm) [inline src/pages/protocol/fuji/cx-tokens.astro:17]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'cxTokens / Neptune Mutual Protocol Contracts on Fuji'
description: 'View all cxToken contracts of the Neptune Mutual Protocol on Fuji'
```

File: src/pages/protocol/fuji/pods.astro [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import { Api } from "../../../../types/enum";
+import HeaderSection from "../../../components/HeaderSection.astro";
+import ProtocolContracts from "../../../views/protocol/ProtocolContracts.astro";
+
+const paginated = await api.getContracts(Api.ContractFuji);
+
+const t = (x: string) => x;
+---
+
+<BaseLayout headerStyle={"colored"}>
```
Comment (binodnpm) [inline src/pages/protocol/fuji/pods.astro:17]: Fix SEO metadata.

- Add OG image first.
- Why no index? This is not correct.

```
title: 'PODs / Neptune Mutual Protocol Contracts on Fuji'
description: 'View all POD contracts of the Neptune Mutual Protocol on Fuji'
```

File: src/pages/protocol/protocol.scss [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+@import "../../components/Header/Navigation.scss";
+@import "../../components/Header/SubMenu.scss";
+@import "../../components/Header/Section.scss";
+@import "../../components/Header/VideoSection.scss";
+@import "../../components/VideoModal.scss";
+@import "../../components/Header/HamburgerButton.scss";
+@import "../../components/CookiePopup.scss";
+@import "../../components/NewsletterSignupForm.scss";
+@import "../../components/EmailSubscription.scss";
+@import "../../components/TableOfContents.scss";
```
Comment (binodnpm) [inline src/pages/protocol/protocol.scss:32]: **Is the file `TableOfContents.scss` really needed?**

File: src/views/homepage/lets-buidl/Features.astro [hunk 12]
```diff
@@ -1,8 +1,13 @@
 ---
 import FeaturedIcon from "../../../elements/FeaturedIcon.astro";
 
-// @todo: Types
-const features = [
+interface FeatureType {
```
Comment (binodnpm) [inline src/views/homepage/lets-buidl/Features.astro:4]: 1. **You should not define inline interfaces.**
2. First, rename the property `content` to `text`.
3. Create a new interface called `FeatureItem` on `custom.d.ts`

```typescript
interface FeatureItem {
  icon: string
  title: string
  text: string
}
```

4. Now, refactor the interface `HomepageFeatureItem` like this:

```typescript
interface HomepageFeatureItem extends FeatureItem {
  id: number
  image: SVGElement
}
```
5. Drop this interface `FeatureType`.
6. Instead, use the newly-created interface `FeatureItem`.
7. Reuse and refactor existing code. Write less code.

File: types/custom.d.ts [hunk 13]
```diff
@@ -184,3 +184,43 @@ interface Sitemap {
...
+  explorer: string
+}
+
+interface NavSection {
+  type: 'section' | 'videos-section'
+  title: string
+  links?: Array<{
+    title: string
+    href: string
+    description: string
```
Comment (binodnpm) [inline types/custom.d.ts:211]: 1. Rename description to text.
2. Use the interface `FeatureItem` here.

File: src/elements/icons/paths.ts [hunk 14]
```diff
@@ -82,5 +84,8 @@ export const paths: Record<string, any> = {
   'metamask-dark': import('./custom/Brands/dark/SvgMetamaskDark.svg?raw'),
   'okx-wallet-dark': import('./custom/Brands/dark/SvgOkxWalletDark.svg?raw'),
   'gnosis-wallet-dark': import('./custom/Brands/dark/SvgGnosisWalletDark.svg?raw'),
-  'wallet-connect-dark': import('./custom/Brands/dark/SvgWalletConnectDark.svg?raw')
+  'wallet-connect-dark': import('./custom/Brands/dark/SvgWalletConnectDark.svg?raw'),
+  'search-lg': import('./variants/General/search-lg.svg?raw'),
+  'link-03': import('./variants/General/link-03.svg?raw'),
+  'arrow-circle-right': import('./variants/Arrows/arrow-circle-right.svg?raw')
 }
```
Comment (binodnpm) [inline src/elements/icons/paths.ts:91]: Please refactor this file to put the `export` directive at the bottom.

**Comment by rupak-nm** [comment]
@binodnpm 
Updated

File: src/pages/protocol/arbitrum/cx-tokens.astro [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+---
+
+<BaseLayout headerStyle={"colored"}>
+  <Seo
+    slot="seo"
+    title="cXTokens / Neptune Mutual Protocol Contracts on Arbitrum"
+    description="View all cxToken contracts of the Neptune Mutual Protocol on Arbitrum"
+    og={{
+      image: "/assets/images/meta/og/protocol-contracts.png",
+      imageAlt: "Neptune Mutual Protocol Contracts Open Graph Banner",
```
Comment (binodnpm) [inline src/pages/protocol/arbitrum/cx-tokens.astro:24]: Make imageAlt same as title on all instances.

File: src/views/protocol/Contract.astro [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+  <div class="icons">
+    <a
+      href={`${network.explorer}${contract.value}`}
+      target="_blank"
+      rel="nofollow"
+    >
+      <Icon variant={"link-03"} />
+    </a>
+
+    <!-- @todo implement the functionality -->
```
Comment (binodnpm) [inline src/views/protocol/Contract.astro:29]: Drop this @todo.

File: src/pages/protocol/arbitrum/cx-tokens.astro (hunk 15 above)
Comment (rupak-nm) [reply src/pages/protocol/arbitrum/cx-tokens.astro]: Done

File: src/views/protocol/Contract.astro (hunk 16 above)
Comment (rupak-nm) [reply src/views/protocol/Contract.astro]: Done

**Review by binodnpm — APPROVED** [review]

**Pull request merged**